### PR TITLE
CPM-816: Identifier Generator Create table to store prefixes

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Automation\IdentifierGenerator\Infrastructure\Symfony\Install;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class CreateIdentifierGeneratorPrefixesTableQuery
```
Comment (pierallard) [inline components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php:11]: IMHO this is not a good idea to create a class just for a const inside.
You should put it back to the original class.
If you find an interest for this, please explain :)

File: components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public const QUERY = <<<SQL
+            CREATE TABLE IF NOT EXISTS pim_catalog_identifier_generator_prefixes (
+                `product_id` INT NOT NULL,
```
Comment (pierallard) [inline components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php:15]: This should be a product_uuid, and this should be a BINARY

File: components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                `attribute_id` INT NOT NULL,
+                `prefix` VARCHAR(255) NOT NULL,
+                `number` INT NOT NULL,
+                CONSTRAINT `FK_PRODUCTID` FOREIGN KEY (`product_id`) REFERENCES `pim_catalog_product` (`id`),
```
Comment (pierallard) [inline components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php:19]: uuid here too

File: upgrades/schema/Version_7_0_20221116131232_add_store_prefixes_identifier_generator_table.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pim\Upgrade\Schema;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version_7_0_20221116131232_add_store_prefixes_identifier_generator_table extends AbstractMigration
```
Comment (pierallard) [inline upgrades/schema/Version_7_0_20221116131232_add_store_prefixes_identifier_generator_table.php:10]: Version_7_0_20221116131232_add_store_prefixes_identifier_generator_table

"store" ?

File: components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php (hunk 1 above)
Comment (antoinearrive) [reply components/identifier-generator/back/src/Infrastructure/Symfony/Install/CreateIdentifierGeneratorPrefixesTableQuery.php]: i saw it here src/Akeneo/Connectivity/Connection/back/Infrastructure/Audit/Install, no problem i have rollback

**Review by Beninho — APPROVED** [review]

**Pull request merged**